channeld/commit_tx.c

```diff
@@ -325,7 +325,7 @@ struct bitcoin_tx *commit_tx(const tal_t *ctx,
 			scriptpubkey = scriptpubkey_p2wpkh(tmpctx,
 							   &keyset->other_payment_key);
 		}
-		pos = bitcoin_tx_add_output(tx, scriptpubkey, NULL, amount);
+		pos = bitcoin_tx_add_output(tx, scriptpubkey, (u8 *) redeem, amount);
```

Comment:
maybe drop the const in the declaration, so you can get rid of the cast here?

(also below)

Reply:
done